providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -706,6 +710,14 @@ static void *dh_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
     if (gctx->group_nid != NID_undef)
         gctx->gen_type = DH_PARAMGEN_TYPE_GROUP;
 
+    /* Bounds check on context gen_type */
+    if (!ossl_assert((gctx->gen_type >= DH_PARAMGEN_TYPE_GENERATOR)
+                    && (gctx->gen_type <= DH_PARAMGEN_TYPE_GROUP))) {
+        ERR_raise_data(ERR_LIB_PROV, ERR_R_INTERNAL_ERROR,
+                       "gen_type set to unsupported value %d", gctx->gen_type);
+        return NULL;
+    }
+
```

Comment:
here we are checking that gen_type is in a valid range (i.e. `lower <= gen_type`  and `gen_type <= upper`).

could you maybe add a comment to `dh.h` that the macro `DH_PARAMGEN_TYPE_GENERATOR` needs to be the lower bound and `DH_PARAMGEN_TYPE_GROUP` needs to be upper bound?  This would defend against the (unlikely) case that someone adds a new TYPE_GROUP macro there.

Reply:
sure